plugins/chanbackup.c

```diff
@@ -267,17 +275,40 @@ static struct command_result *json_emergencyrecover(struct command *cmd,
 				    forward_error, NULL);
 
 	json_array_start(req->js, "scb");
-	for (size_t i=0; i<tal_count(scb); i++) {
-		u8 *scb_hex = tal_arr(cmd, u8, 0);
-		towire_scb_chan(&scb_hex,scb[i]);
-		json_add_hex(req->js, NULL, scb_hex, tal_bytelen(scb_hex));
+	if (is_tlvs) {
+		for (size_t i=0; i<tal_count(scb_tlvs); i++) {
+			u8 *scb_hex = tal_arr(cmd, u8, 0);
+			towire_scb_chan_with_tlvs(&scb_hex,scb_tlvs[i]);
+			json_add_hex(req->js, NULL, scb_hex, tal_bytelen(scb_hex));
+		}
+	} else {
+		for (size_t i=0; i<tal_count(scb); i++) {
+			plugin_log(cmd->plugin, LOG_DBG,
+				  "Processing legacy emergency.recover file format. "
+				  "Please migrate to the latest file format for improved "
+				  "compatibility and fund recovery.");
```

Comment:
You can use "json_notify_fmt()" to send a notification to the caller (though maybe only log that once,not every time around the loop!).

Reply:
used plugin_notify_message.